core/cont/inc/TBits.h

```diff
@@ -20,7 +20,6 @@
 //                                                                      //
 //////////////////////////////////////////////////////////////////////////
 
-#include "Rtypes.h"
```

Comment:
`UChar_t` is used in this file and I think we should explicitly include `Rtypes.h` and not rely on the transitive dependency coming from `TObject.h`.

I would recommend installing and configuring include-what-you-use. This would make this work sustainable: it won't regress easily; and it will help us detect such issues.

Reply:
`Rtypes.h` includes in `TObject.h`. But I agree - one can can it here

Actually, I tried to use `include-what-you-use` several times, but always got problem with my python environment.

Reply:
From other point of view - if `TObject.h` is included, it is guaranteed that `Rtypes.h` is also there. Like in several places I removed `TObject.h` when already `TNamed.h` or any other derived class already included.

Reply:
Your logic - more or less in every place where `Int_t` or `UInt_t` is used one have to place `#include "RtypesCore.h"`. Or in 99% of ROOT source files. In my mind - it is overkill.

Reply:
That allows to change includes in things such as TObject.h without breaking code. It also allows better layering and actually allows building module files (-fmodules) independently. If something has only Int_t why including TObject.h or anything else rather the header defining Int_t. In that case the built module will not depend on the ROOT’s code module which contains TObject.h

Reply:
But I see no chance that `Rtypes.h` include will be removed from `TObject.h`. And taking this as fact - it makes no sense to keep them both. 

This PR - try to minimize includes in header files. If we want to split some basic includes like `TObject.h` - one can discuss this, but this will be absolutely different PRs.

Reply:
Just to make a few steps back. Do we want to reduce the number of includes or we want to include only what we use?

If the former, then we put all core includes in one file and include it, if the latter these sort of changes does not bring us any closer.

Reply:
> Just to make a few steps back. Do we want to reduce the number of includes or we want to include only what we use?

I guess, main idea to reduce existing includes to have only that is really used. While we cannot simply remove all unused includes - move under `R__LESS_INCLUDES` define.